### PR TITLE
Drop Vtk from conda-env

### PR DESCRIPTION
**Pull request by inducer**
Vtk pulls in a lot of heavyweight dependencies. Its only purpose at this point seems to be to enable `mirgecompare`, but that doesn't justify a few extra gigabytes and a few extra minutes of install time, at least to me.

@matthiasdiener How do you feel about that? `mirgecompare` could simply have instructions to `conda install vtk` if the relevant packages are missing. Happy to add that if you agree.

Production changes to be merged alongside this:

https://github.com/illinois-ceesd/mirgecom/compare/production...production-merge/drop-vtk-from-env

### CODE REVIEW

**Comment by inducer** [comment]
@matthiasdiener I'll wait for your feedback before sorting out the production merge failure.

**Comment by matthiasdiener** [comment]
Imho, if it is only needed for mirgecompare, dropping `vtk` is a good idea.

**Review by matthiasdiener — APPROVED** [review]

File: bin/mirgecompare.py
Comment (matthiasdiener) [inline bin/mirgecompare.py:1]: Most things installed through emirge seem to have `#!/usr/bin/env python` but I'm agnostic on this.

**Comment by inducer** [comment]
@MTCam How bad would it be to merge this? AFAICT, this would only cause a trivial merge conflict in `conda-env.yml`.

File: bin/mirgecompare.py
Comment (inducer) [reply bin/mirgecompare.py]: Fair. Reverted for consistency.

**Comment by MTCam** [comment]
> @MTCam How bad would it be to merge this? AFAICT, this would only cause a trivial merge conflict in `conda-env.yml`. (If it's bad, what's the suggested procedure for getting this in?)

I apologize for taking so long to get back with you about this.   I only just now noticed this mention.   This change set is pretty innocuous.  I'd say merge at will.

FWIW: If it did have the potential to affect the production cases, then the path for making it production-ready is outlined in the docs:  https://mirgecom.readthedocs.io/en/latest/development/pullrequests.html#production-testing-in-ci

**Comment by inducer** [comment]
The main issue is that the production branch doesn't merge cleanly. I'm not sure the document you point to addresses the situation of a merge conflict and/or how to address it.

**Comment by MTCam** [comment]
> The main issue is that the production branch doesn't merge cleanly. I'm not sure the document you point to addresses the situation of a merge conflict and/or how to address it.

This is going to sound like an RTFM-type answer, but that is not my intent.     

The first point I want to make before continuing; it is not strictly necessary to resolve this before merging.   Merging stuff that breaks production tests makes me grumpy, but it is not a giant deal.  If the merge is simple, and does not hurt production, then production will pick it up when it updates to pick up the new main (typically as soon as I see production tests on main start failing).   

Now for the rtfm part: Steps 1 and 2 outlined in the documentation above are specifically designed to discover and deal with merge conflicts and offer a path to resolution.

In summary of steps 1 and 2: 
- Check out the branch
- merge production branch (resolve conflicts yourself)
- push the resulting code as your "custom" production branch 
- specify the custom production branch in your production testing env for the branch

This procedure is typically only required for developments that deeply affect production.

For developments such as yours, that don't affect production at all, you can just merge with no worries and let me deal with it, or you can pre-merge your development to production and viola: merge conflicts resolved.   Not only does a pre-merge to production fix your own development's merge conflicts, but it also let's main test out production tests with your changes applied before you merge them.  These two paths to resolving merge conflicts before merge-to-main are really identical.  If you are confident that your changes don't hurt production - then pre-merge is best as it avoids production env customization.   Again, for developments that profoundly affect production code - the best path is production env customization as outlined above.

I am more than happy to help with this, and beef up the documentation if you find it lacking.

**Comment by inducer** [comment]
Thanks for explaining. I'd like to propose an improvement to that process, which I've realized in 5e3ce201. The main changes are that:

- There's a naming convention for production branches, namely `production-merge/MAIN_BRANCH_NAME`.
- The production testing branch switch deactivates itself automatically once merged to main, so that it doesn't need to be backed out before merging. (And, potentially, there could be a on-push Github workflow that automatically resets production to the relevant branch.)

@matthiasdiener @MTCam LMK what you think of this. I've added it to this PR as a test, I'm happy to separate it out.

:warning: If we do merge as part of this, it please don't squash, so that unrelated changes don't get mashed into the same commit. 

L

**Comment by MTCam** [comment]
> Thanks for explaining. I'd like to propose an improvement to that process, which I've realized in [5e3ce20](https://github.com/illinois-ceesd/mirgecom/commit/5e3ce201e251de56888e9fbe33b50def1ff8ee6a). The main changes are that:
> 
> * There's a naming convention for production branches, namely `production-merge/MAIN_BRANCH_NAME`.

I like this.  It'd be great if it could be automatically deleted MAIN_BRANCH_NAME merge to main.  Alas, probably too much to ask for.

> * The production testing branch switch deactivates itself automatically once merged to main, so that it doesn't need to be backed out before merging. (And, potentially, there could be a on-push Github workflow that automatically resets production to the relevant branch.)

This would be terrific, if it works!

> 
> @matthiasdiener @MTCam LMK what you think of this. I've added it to this PR as a test, I'm happy to separate it out.
> 
> ⚠️ If we do merge as part of this, it please don't squash, so that unrelated changes don't get mashed into the same commit.

**Comment by inducer** [comment]
@matthiasdiener Could you look over my shell scripting to see if there's something fishy?

File: .ci-support/merge-install-production-branch.sh
Comment (MTCam) [inline .ci-support/merge-install-production-branch.sh:27]: This seems to indicate PRs *must* have a compatible production branch instead of allowing a default to regular ole production.   If it merges, it seems it will break all production feeder branches - none of which require such a custom production branch.

File: .ci-support/merge-install-production-branch.sh
Comment (matthiasdiener) [inline .ci-support/merge-install-production-branch.sh:37]: Could mention here where $CI is set

File: .ci-support/merge-install-production-branch.sh
Comment (matthiasdiener) [inline .ci-support/merge-install-production-branch.sh:59]:
```suggestion
  else
```

**Comment by matthiasdiener** [comment]
> @matthiasdiener Could you look over my shell scripting to see if there's something fishy?

Looks good, except the two minor comments I made. I think MikeC's comment still needs addressing though.

File: .ci-support/merge-install-production-branch.sh
Comment (inducer) [reply .ci-support/merge-install-production-branch.sh:27]: That's not the intention. What makes you think that? To me, the logic is that "if we're currently building a PR *and* if the production env settings indicate that a special branch is needed (by setting `PR_MAIN_BRANCH` to the name of the current branch... come to think of it, I'll probably rename that variable, likely `PRODUCTION_PR_MAIN_BRANCH`), then use that branch. Otherwise, fall back to the old logic.

File: .ci-support/merge-install-production-branch.sh
Comment (inducer) [reply .ci-support/merge-install-production-branch.sh]: https://github.com/illinois-ceesd/mirgecom/compare/5e3ce201e251de56888e9fbe33b50def1ff8ee6a..e3725cd0d5cfb267c7deb3b0906977a0c38d5bf9

File: .ci-support/merge-install-production-branch.sh
Comment (inducer) [reply .ci-support/merge-install-production-branch.sh:39]: https://github.com/illinois-ceesd/mirgecom/compare/5e3ce201e251de56888e9fbe33b50def1ff8ee6a..e3725cd0d5cfb267c7deb3b0906977a0c38d5bf9

File: .ci-support/merge-install-production-branch.sh
Comment (MTCam) [reply .ci-support/merge-install-production-branch.sh]: The previous logic was if the PR requires a special production branch, then they set PRODUCTION_BRANCH.  I am unsure why additional variables are needed.

Comment (inducer) [reply .ci-support/merge-install-production-branch.sh]: No, the original variables stay unchanged. The additional variable (now `PRODUCTION_PR_MAIN_BRANCH`) contains the name of the branch headed into `main`. That way, the CI script can tell whether it's currently testing that branch and, if so, test the corresponding production branch (found from `PRODUCTION_PR_MAIN_BRANCH` by convention) by appropriately populating the existing variables. Likewise, `PRODUCTION_PR_FORK` is needed because it's only supposed to be active *if* we're still on the specific branch.

Comment (MTCam) [reply .ci-support/merge-install-production-branch.sh]: I do trust you know what you are doing; I just don't understand.   

When CI runs these scripts, it is always testing the current branch, so I didn't find it necessary to indicate to CI what branch it is currently testing.   The production testing procedure was to merge "production" into the current branch (i.e. the branch being tested, regardless of what branch it is)  and run the production cases, and any branch could specify a custom branch to use for "production" through the PRODUCTION_BRANCH variable.

It appeared that this change set removes the PRODUCTION_BRANCH variable hooks from `production_testing_env.sh`, and force the setting of PRODUCTION_PR_MAIN_BRANCH.   The previous logic would use `production` proper iff the PRODUCTION_BRANCH var was not set.

File: .ci-support/merge-install-production-branch.sh
Comment (MTCam) [inline .ci-support/merge-install-production-branch.sh:27]: I think the `else` is unneeded.  Use what was set just above or default to production proper.

```suggestion
PRODUCTION_BRANCH="${PRODUCTION_BRANCH:-"production"}"
PRODUCTION_FORK=${PRODUCTION_FORK:-"illinois-ceesd"}
```

Comment (inducer) [reply .ci-support/merge-install-production-branch.sh:27]: I'm not sure I understand what you're asking for here. Your first comment seems to aim at removing `PRODUCTION_{FORK,BRANCH}`, and your second one seems to push for reinstating them. Which of the two would you like? And if you're interested in reinstating them, for what use case that isn't a PR?

Comment (MTCam) [reply .ci-support/merge-install-production-branch.sh:27]: This change set forces a custom production branch called: `production-merge/${PR_MAIN_BRANCH}`.   I am accustomed to calling my custom production branch whatever I want:
`PRODUCTION_BRANCH=my-custom-production-branch`.    

I would like to retain that capability.   Nothing in this change set prevents my continuing to do that, except the comment snippet that indicates something like that is possible has been removed from `production_test_env.sh`.  I am merely suggesting that snippet should be restored.

Comment (inducer) [reply .ci-support/merge-install-production-branch.sh:27]: > I am merely suggesting that snippet should be restored.

If it's restored, wouldn't that confuse new users who might see two ways to do the same thing and be confused on which of them they should use? (To my mind, the one with the naming convention is preferable, so I tried to make that the path of least resistance.)

Comment (MTCam) [reply .ci-support/merge-install-production-branch.sh:27]: To me it isn't the same thing.  It simply allows a user to specify outright the name of the custom production branch to use at their option.

With the new way you are introducing, one is forced to have an independent custom production for each branch/PR, with the advantage that it does not need to be "undone" before merge.   In the way that I typically operate, I can and do use a single custom production branch for several developments/branches/PRs.   I would like to retain that ability.

**Pull request closed without merging**
